src/Functions/IFunction.cpp

```diff
@@ -285,6 +285,8 @@ ColumnPtr IExecutableFunction::executeWithoutSparseColumns(const ColumnsWithType
                 ? res->cloneResized(1)->convertToFullColumnIfConst()
                 : res;
 
+            keys = keys->convertToFullColumnIfLowCardinality();
```

Comment:
A couple of lines above `IFunction::executeWithoutLowCardinalityColumns` is called with `columns_without_low_cardinality`. `columns_without_low_cardinality` indeed contains unwrapped columns (as in not LC), but the value returned by `IFunction::executeWithoutLowCardinalityColumns` is Const(ColumnLowCardinality), which begs the question if that's the expected behavior. Doesn't seem like.

`res` is then used to create the keys, which is passed to `res_mut_dictionary->uniqueInsertRangeFrom` that expects an unwrapped LC column.

Calling `keys->convertToFullColumnIfLowCardinality` fixes the problem, but I wonder if that's the right way to fix it. My instinct tells me `executeWithoutLowCardinalityColumns` is the actual problem, but unfortunately my debugger won't allow me to properly debug this file. Hence, I am posting this comment as a quesiton.

What is the expected behavior/ return value of `executeWithoutLowCardinalityColumns`?

Reply:
Hi @Avogar, since you are an assignee, could you please clarify ^^ ?

Reply:
Yes, you are right, it's not an expected behaviour of `executeWithoutLowCardinalityColumns`. The problem is in function `__getScalar` (here)(https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/getScalar.cpp) it should not use default implementation for LC columns. So we should add
```c++
bool useDefaultImplementationForLowCardinalityColumns() const override { return false; }
```